src/dmd/cparse.d

```diff
@@ -3179,8 +3171,14 @@ final class CParser(AST) : Parser!AST
             if (!symbols)
                 symbols = new AST.Dsymbols;     // lazilly create it
 
-            if (!tspec && !specifier.scw && !specifier.mod)
+            if (!tspec && !specifier.scw && !specifier.mod && !width)
                 error("specifier-qualifier-list required");
```

Comment:
I'm not sure this is correct.  You want an error if there is no type for the bit-field.  Otherwise it would appear that you are allowing the following to be valid.
```
struct foo
{
    : 32;
};
```

Reply:
Trying that code gives:
```
Error: empty struct-declaration-list for `struct S`
```

Reply:
fixed